sdk/src/main/java/com/kin/ecosystem/marketplace/presenter/MarketplacePresenter.java

```diff
@@ -83,17 +85,21 @@ public void onAttach(IMarketplaceView view) {
 	}
 
 	private void getCachedOffers() {
+		OfferList cachedOfferList = offerRepository.getCachedOfferList();
 		if (earnList == null && spendList == null) {
 			earnList = new ArrayList<>();
 			spendList = new ArrayList<>();
-			OfferList cachedOfferList = offerRepository.getCachedOfferList();
-			if (cachedOfferList != null && cachedOfferList.getOffers() != null) {
-				OfferListUtil.splitOffersByType(cachedOfferList.getOffers(), this.earnList, this.spendList);
-			}
+		}
+		if (hasOffers(cachedOfferList)) {
```

Comment:
no need for if (hasOffers(cachedOfferList)) it is also checked in  syncOffers

Reply:
👍